src/integer.rs

```diff
@@ -475,3 +510,154 @@ impl TryConvert for Integer {
         }
     }
 }
+
+impl PartialEq for Integer {
+    fn eq(&self, other: &Self) -> bool {
+        match self.integer_type() {
+            IntegerType::Bignum(a) => unsafe {
+                rb_sys::rb_big_eq(a.as_rb_value(), other.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+            IntegerType::Fixnum(a) => a.as_rb_value() == other.norm().as_rb_value(),
+        }
+    }
+}
+
+impl PartialOrd for Integer {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.binary_operation_visit(
+            other,
+            |a, b| (a.as_rb_value() as i64).partial_cmp(&(b.as_rb_value() as i64)),
+            |a, b| unsafe {
+                let result = rb_sys::rb_big_cmp(a, b);
+                Integer::from_rb_value_unchecked(result)
+                    .to_i64()
+                    .unwrap()
+                    .partial_cmp(&0)
+            },
+        )
+    }
+}
+
+impl Add for Integer {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        self.binary_operation_visit(
+            &other,
+            |a, b| {
+                let raw_a = a.as_rb_value() as i64;
+                let raw_b = b.as_rb_value() as i64;
```

Comment:
I think this needs to be `isize` to support 32 bit platforms?

Reply:
It's [complicated](https://doc.rust-lang.org/src/core/ffi/mod.rs.html#179)... but wouldn't this need to be `c_long` due to `RUBY_FIXNUM_MAX` being defined as a `c_long`?

Reply:
oh yeah, I think you're right. And because `c_long` is always 32bits on Windows, Windows has a lower max fixnum even on 64bit systems.